v1/common/amqp.go

```diff
@@ -111,6 +131,54 @@ func (ac *AMQPConnector) Open(url string, tlsConfig *tls.Config) (*amqp.Connecti
 	return conn, channel, nil
 }
 
+func (ac *AMQPConnector) getConn(url string, tlsConfig *tls.Config, keepAlive bool) (*amqp.Connection, error) {
+	ac.mu.Lock()
+	defer ac.mu.Unlock()
+
+	if !keepAlive {
+		return ac.createNewConn(url, tlsConfig)
+	}
+
+	var done, makeNew bool
+	for !done {
+		select {
+		case err := <-ac.connChan:
+			if err == nil {
+				done = true
+			}
+			makeNew = true
+		default:
+			done = true
+		}
+	}
+	if ac.conn != nil && !makeNew {
+		return ac.conn, nil
+	}
+
+	conn, err := ac.createNewConn(url, tlsConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	ac.setConn(conn)
+	return conn, nil
+}
+
+func (ac *AMQPConnector) createNewConn(url string, tlsConfig *tls.Config) (*amqp.Connection, error) {
+	// From amqp docs: DialTLS will use the provided tls.Config when it encounters an amqps:// scheme
+	// and will dial a plain connection when it encounters an amqp:// scheme.
+	conn, err := amqp.DialTLS(url, tlsConfig)
+	if err != nil {
+		return nil, fmt.Errorf("Dial error: %s", err)
+	}
+	return conn, nil
+}
+
+func (ac *AMQPConnector) setConn(conn *amqp.Connection) {
+	ac.conn = conn
+	conn.NotifyClose(ac.connChan)
```

Comment:
According to the docs NotifyClose will close the channel if the connection is closed with a normal shutdown. So, I think a new channel should be created here each time.

Reply:
Oh yeah, I missed it.